pol-core/bscript/bclassinstance.cpp

```diff
@@ -0,0 +1,120 @@
+#include "bclassinstance.h"
+
+#include "berror.h"
+#include "bobject.h"
+#include "objmethods.h"
+
+namespace Pol::Bscript
+{
+BClassInstance::BClassInstance( ref_ptr<EScriptProgram> program, int index,
+                                std::shared_ptr<ValueStackCont> globals )
+    : BStruct( OTClassInstance ), prog_( program ), index_( index ), globals( std::move( globals ) )
+{
+}
+
+BClassInstance::BClassInstance( const BClassInstance& B ) : BStruct( B, OTClassInstance )
+{
+  prog_ = B.prog_;
+  index_ = B.index_;
+  globals = B.globals;
+}
+
+size_t BClassInstance::sizeEstimate() const
+{
+  return sizeof( BClassInstance ) + prog_->sizeEstimate();
```

Comment:
I don't think you have to add the complete prog here. That would be only true if the class is the last one holding the program